google_fonts/lib/src/google_fonts_base.dart

```diff
@@ -0,0 +1,190 @@
+  final path = await _localPath;
+  return File('$path/$name.ttf');
+}
+
+Future<File> _writeLocalFont(String name, List<int> bytes) async {
+  final file = await _localFile(name);
+  return file.writeAsBytes(bytes);
+}
+
+Future<ByteData> _readLocalFont(String name) async {
+  try {
+    final file = await _localFile(name);
+    final fileExists = await file.exists();
+    if (fileExists) {
+      List<int> contents = await file.readAsBytes();
+      if (contents != null && contents.isNotEmpty) {
+        return ByteData.view(Uint8List.fromList(contents).buffer);
+      }
+    }
+  } catch (e) {
+    return null;
+  }
+  return null;
+}
+
+String _fontWeightString(TextStyle textStyle) {
+  if (textStyle.fontWeight == null || textStyle.fontWeight == FontWeight.w400) {
+    return '';
+  }
+  return textStyle.fontWeight.toString().replaceAll('FontWeight.w', '');
+}
+
+String _fontStyleString(FontStyle fontstyle) {
+  if (fontstyle == FontStyle.italic) {
+    return 'italic';
+  }
+  return '';
+}
+
+class GoogleFontsFamily {
+  const GoogleFontsFamily(this.fontWeight, this.fontStyle, [this.url]);
+
+  // The FontWeight as an index [0-8] representing the font weights 100-900.
+  final int fontWeight;
+  final FontStyle fontStyle;
+  final String url;
+
+  // From a string key, for example, `500italic`.
+  GoogleFontsFamily.fromString(String key, [this.url])
+      : this.fontWeight = key == _regular || key == _italic
+            ? 3
+            : (int.parse(key.replaceAll(_italic, '')) ~/ 100) - 1,
+        this.fontStyle =
+            key.contains(_italic) ? FontStyle.italic : FontStyle.normal;
+
+  GoogleFontsFamily.fromTextStyle(TextStyle style, [this.url])
+      : this.fontWeight = style?.fontWeight?.index ?? 3,
+        this.fontStyle = style?.fontStyle ?? FontStyle.normal;
+
+  @override
+  String toString() {
+    final fontWeightString = (fontWeight + 1) * 100;
+    final fontStyleString = fontStyle.toString().replaceAll('FontStyle.', '');
+    return '$fontWeightString$fontStyleString';
+  }
+
+  @override
+  int get hashCode => hashValues(fontWeight, fontStyle);
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    final GoogleFontsFamily typedOther = other;
+    return typedOther.fontWeight == fontWeight &&
+        typedOther.fontStyle == fontStyle;
+  }
+}
+
+// This logic is taken from the following section of the minikin library, which
+// is ultimately how flutter handles matching fonts.
+// * https://github.com/flutter/engine/blob/master/third_party/txt/src/minikin/FontFamily.cpp#L128
+int _computeMatch(GoogleFontsFamily a, GoogleFontsFamily b) {
+  if (a == b) {
+    return 0;
+  }
+  int score = (a.fontWeight - b.fontWeight).abs();
+  if (a.fontStyle != b.fontStyle) {
+    score += 2;
+  }
+  return score;
+}
+
+const _regular = 'regular';
+const _italic = 'italic';
```

Comment:
use or remove this constants.

Reply:
They are being used in GoogleFontsFamily.fromString